examples/material/create_material.py

```diff
@@ -1,28 +1,69 @@
 #!/usr/bin/env python
 # coding: utf-8
 
+# [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Exabyte-io/exabyte-api-examples/blob/feature/SOF-4400-skinny-req/examples/material/create_material.ipynb)
+
 # # Overview
 # 
 # In this example we create a material from a JSON config with [tags](https://docs.exabyte.io/entities-general/data/#tags) to identify the material.
 
-# # Execution
-# 
-# > <span style="color: orange">**NOTE**</span>: In order to run this example, an active Exabyte.io account is required. RESTful API credentials shall be updated in [settings](../settings.py). The generation of the credentials is also explained therein.
+# # Complete Authorization Form and Initialize Settings
 # 
-# ## Import packages
+# This will also determine environment and set all environment variables. We determine if we are using Jupyter Notebooks or Google Colab to run this tutorial. If google drive, we will have some extra steps to do.
 
-# In[]:
+# In[ ]:
 
 
+#@title Authorization Form
+ACCOUNT_ID = "" #@param {type:"string"}
+AUTH_TOKEN = "" #@param {type:"string"}
+MATERIALS_PROJECT_API_KEY = "" #@param {type:"string"}
+
 import os
-import sys
 
-# Import settings and utils file
+if 'google.colab' in str(get_ipython()):
+  print('Running on CoLab')
+  os.environ['notebook_environment'] = "Colab"
+elif 'ZMQInteractiveShell'  in str(get_ipython()):
+  print('Running in Jupyter')
+  os.environ['notebook_environment'] = "Jupyter"
+else:
+  print('Unknown Environment')
+  os.environ['notebook_environment'] = ""
+    
+if os.environ['notebook_environment'] == "Colab":
+    get_ipython().system('git clone -b feature/SOF-4400-skinny-req https://github.com/Exabyte-io/exabyte-api-examples.git')
+    get_ipython().run_line_magic('cd', '/content/exabyte-api-examples/examples/material')
```

Comment:
We could get the `/examples/material` automatically from the current file name

Reply:
Hi Timur,

In Colab, our initial location upon instantiation is: '/content'. I haven't been able to think of anyways we can get the path to where the notebook is inside the repo once it is cloned.

Are you OK if we leave /examples/material  here?

One thing I could remove is /content, since we are already there, it is redundant.

Reply:
My point was to move the lines 24-50 to a util function. We could pass the current file as argument to this util function maybe.

Reply:
Thanks Timur.

I don't think we can place the lines that detect the notebook environment (lines 24-35) into utils because we would need to know if we were in Colab first before we clone the repo.

I unfortunately think line 36 will need to stay. I am not finding a way to get the information about where the notebook is.

Line 37 could be moved into a function, would you want me to do that still?

Lines 39-50 have been moved into a utils function -> utils.generic.update_settings_with_users_authorization_info(..)